packages/template-retail-react-app/app/utils/routes-utils.js

```diff
@@ -9,25 +9,99 @@ import {getSites} from '@salesforce/retail-react-app/app/utils/site-utils'
 import {urlPartPositions} from '@salesforce/retail-react-app/app/constants'
 
 /**
- * Construct literal routes based on url config
- *      with site and locale references (ids and aliases) from each in your application config
+ * Build regex patterns for site and locale route parameters.
+ * Creates patterns like "siteA|siteB|siteC" from all valid site/locale refs.
+ *
+ * @param {array} allSites - array of site configurations
+ * @returns {object} - { sitePattern, localePattern }
+ */
+const buildRoutePatterns = (allSites) => {
+    const siteRefs = allSites.flatMap((site) => [site.alias, site.id]).filter(Boolean)
+
+    const localeRefs = allSites
+        .flatMap((site) => site.l10n.supportedLocales)
+        .flatMap((locale) => [locale.alias, locale.id])
+        .filter(Boolean)
+
+    // Remove duplicates and join into regex pattern
+    const sitePattern = [...new Set(siteRefs)].join('|')
```

Comment:
nit, could `siteRefs` contain regex sensitive characters like  (`.`, `+`, `*`, `(`, etc.) and cause regex matching issues? like `site.uk`, should we sanitize and escape those chars?

Reply:
I think that comment applies to the site id locale id since what is what those values are. I don't think we are currently validating those values in our code as of yet. Although I con't think there will be too many customer using `+` as a site id. It's possible, but a fringe case. This should be something we validate in Storefront Next tho.